src/poller.ts

```diff
@@ -1,35 +1,134 @@
+import {
+  DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES,
+  DEFAULT_POLL_CONFIG_REQUEST_RETRIES,
+  POLL_JITTER_PCT,
+} from './constants';
+
 export interface IPoller {
   start: () => Promise<void>;
   stop: () => void;
 }
 
 export default function initPoller(
-  interval: number,
+  intervalMs: number,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   callback: () => Promise<any>,
+  options?: {
+    maxPollRetries?: number;
+    maxStartRetries?: number;
+    errorOnFailedStart?: boolean;
+    pollAfterFailedStart?: boolean;
+  },
 ): IPoller {
   let stopped = false;
+  let failedAttempts = 0;
+  let nextPollMs = intervalMs;
+
+  const start = async () => {
+    stopped = false;
+    let startRequestSuccess = false;
+    let startAttemptsRemaining =
+      1 + (options?.maxStartRetries ?? DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES);
+
+    let startError = null;
+
+    while (!startRequestSuccess && startAttemptsRemaining > 0) {
+      try {
+        await callback();
+        startRequestSuccess = true;
+      } catch (pollingError) {
+        console.warn(
+          `Eppo SDK encountered an error with initial poll of configurations: ${pollingError.message}`,
+        );
+        if (--startAttemptsRemaining > 0) {
+          const jitterMs = randomJitterMs(intervalMs);
```

Comment:
This _may_ be overkill, but when retrying things, including jitter is a best practice in case the issues are from an overwhelmed resource.

Reply:
Jitter is not overkill. If timeout is due to too many simultaneous requests, the issue will never resolve if all subsequent retries also all happen simultaneously, so it's crucial in some scenarios